### PR TITLE
Delete users created by MySQL Router charm when relation broken

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**
## Issue
Previously in https://github.com/canonical/mysql-router-k8s-operator/pull/51, the mysql-router-k8s charm deleted users it created during the relation broken event (for the relation to MySQL). This led to a race condition, since on the relation broken event the MySQL charm deletes the user that the router was using to delete users

## Solution
Delete users created by the MySQL Router charm in the MySQL charm's relation broken hook

### CODE REVIEW

**Review by shayancanonical — COMMENTED** [review]
Left some comments in the VM charm PR (#207). Additionally, I think we need to migrate fixes for Rados gateway to this charm from the VM charm before we can merge these changes safely

File: lib/charms/mysql/v0/mysql.py
Comment (shayancanonical) [reply lib/charms/mysql/v0/mysql.py:1361]: i have a ticket (dpe-1891) for this pulse to update the lib(s) for backup. i will prioritize these changes today in order to unblock you

**Review by paulomach — COMMENTED** [review]
discussion on k8s charm, but I will use this as hook - @carlcsaposs-canonical it's generally more effective to approve on a single repo before porting.

**Review by delgod — APPROVED** [review]

**Review by taurus-forever — COMMENTED** [review]
LGTM, but please sync with Shayan and Paulo about merging order (re: backup backporing fixes).

**Review by delgod — APPROVED** [review]

**Review by shayancanonical — APPROVED** [review]

**Review by paulomach — APPROVED** [review]
incredible

**Pull request merged**
